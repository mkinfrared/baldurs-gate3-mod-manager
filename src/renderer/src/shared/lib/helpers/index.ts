export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
