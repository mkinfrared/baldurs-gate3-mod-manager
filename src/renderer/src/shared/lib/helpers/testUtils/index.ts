export { renderWithRouter, withQueryProvider } from "./testUtils";
