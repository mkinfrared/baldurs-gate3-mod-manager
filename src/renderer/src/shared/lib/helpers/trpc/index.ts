export { trpc, trpcClient } from "./trpc";
