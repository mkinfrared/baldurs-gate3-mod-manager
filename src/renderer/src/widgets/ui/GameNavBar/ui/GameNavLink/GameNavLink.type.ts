import { ReactNode } from "react";
import { LinkProps } from "react-router-dom";

export type GameNavLinkProps = {
  children?: ReactNode;
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  to: LinkProps["to"];
};
