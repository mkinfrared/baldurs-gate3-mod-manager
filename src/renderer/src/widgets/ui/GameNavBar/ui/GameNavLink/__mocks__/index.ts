export { GameNavLinkMock as GameNavLink } from "./GameNavLink.mock";
