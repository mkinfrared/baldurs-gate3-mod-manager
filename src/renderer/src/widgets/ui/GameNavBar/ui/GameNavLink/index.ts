export { GameNavLink } from "./GameNavLink";

export * from "./GameNavLink.type";
