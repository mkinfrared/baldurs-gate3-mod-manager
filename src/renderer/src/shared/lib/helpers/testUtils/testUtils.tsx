/* eslint-disable import/no-unused-modules */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { ReactElement, ReactNode } from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { trpc, trpcClient } from "@renderer/shared/lib/helpers/trpc";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

const queryClient = new QueryClient();

const withQueryProvider = (children: ReactNode) => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </trpc.Provider>
);

const withRouter = (children: ReactNode) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export { renderWithRouter, withQueryProvider, withRouter };
