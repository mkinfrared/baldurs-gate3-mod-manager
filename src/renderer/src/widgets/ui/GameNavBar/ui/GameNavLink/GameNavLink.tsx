import { NavLink as RouterLink, useMatch } from "react-router-dom";

import { classNames } from "@renderer/shared/lib/helpers";
import { Link } from "@renderer/shared/ui";

import css from "./GameNavLink.module.scss";
import { GameNavLinkProps } from "./GameNavLink.type";

const GameNavLink = ({ children, className, to }: GameNavLinkProps) => {
  const match = useMatch(to.toString());
  const variant = match ? "contained" : "outlined";

  return (
    <Link
      as="icon"
      className={classNames(css.GameNavLink, className)}
      component={RouterLink}
      to={to}
      variant={variant}
    >
      {children}
    </Link>
  );
};

export { GameNavLink };
