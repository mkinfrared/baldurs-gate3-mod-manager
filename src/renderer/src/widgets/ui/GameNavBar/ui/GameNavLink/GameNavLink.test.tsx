import Bell from "@renderer/shared/assets/icons/bell.svg";
import { renderWithRouter } from "@renderer/shared/lib/helpers/testUtils";

import { GameNavLink } from "./GameNavLink";

describe("<GameNavLink />", () => {
  const to = "/foobar";

  const Component = (
    <GameNavLink to={to}>
      <Bell />
    </GameNavLink>
  );

  it("should be defined", () => {
    expect(GameNavLink).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = renderWithRouter(Component);

    expect(container).toMatchSnapshot();
  });
});
