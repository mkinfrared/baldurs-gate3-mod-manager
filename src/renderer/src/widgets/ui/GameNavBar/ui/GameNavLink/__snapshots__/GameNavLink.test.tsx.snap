// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GameNavLink /> > should match the snapshot 1`] = `
<div>
  <a
    class="Link GameNavLink outlined primary icon"
    data-testid="Link"
    href="/foobar"
  >
    <svg>
      bell.svg
    </svg>
  </a>
</div>
`;
